crates/core/component/dex/src/component/position_manager.rs

```diff
@@ -462,28 +463,28 @@ impl<T: StateWrite + ?Sized + Chandelier> PositionManager for T {}
 trait Inner: StateWrite {
     /// Writes a position to the state, updating all necessary indexes.
     ///
-    /// This should be the SOLE ENTRYPOINT for writing positions to the state.
+    /// This should be the **SOLE ENTRYPOINT** for writing positions to the state.
     /// All other position changes exposed by the `PositionManager` should run through here.
     #[instrument(level = "debug", skip_all)]
     async fn update_position(
         &mut self,
+        id: &position::Id,
```

Comment:
why is it bad for this fn to re-hash the position ID? it seems like as implemented it is subject to misuse where you could specify the wrong position ID for a position you're updating

Reply:
The motivation is that doing this on every single position state update adds up